src/coreclr/vm/interpexec.cpp

```diff
@@ -3325,6 +3343,19 @@ do {                                                                           \
                     break;
                 }
 
+                case INTOP_CONV_NI:
+                {
+                    NULL_CHECK(LOCAL_VAR(ip[2], void*));
+                    int64_t index = LOCAL_VAR(ip[3], int64_t);
+                    if ((index < 0) || (index > UINT_MAX))
```

Comment:
Isn't the max legal array index `0x7FEFFFFF`?

Reply:
Yes, but the normal index check will check that one.